### DISCUSSION TITLE
Tasmota devices Unavailable after Home Assistant crash

### DISCUSSION THREAD

**Discussion by sillyfrog** [arendst/Tasmota#12142] [Q&A] [answered]
I have seen others hint at the problem, but no actual fix. I have an issue where if my Home Assistant docker container crashes, all of my Tasmota devices show up as "Unavailable". (A clean restart is OK, but a hard HA crash appears to cause the issue.) I'm using the mosquitto broker.

I'm using the Tasmota Home Assistant native integration with Tasmota versions v9.3.0 and v9.3.1. I have also just updated Home Assistant to 2021.5.5.

I can get them to show up again as available in HA by restarting the devices - but that's starting to be a lot of devices.

I have left it for over 20 hours, and _none_ of the devices have come back online (unless I reboot them).

I tried doing a manual publish of an online message, ie:
```
mosquitto_pub -t 'tele/machine-dehumidifier/LWT' -m 'Online'
```
But that didn't help (I also confirmed that the message is retained).

Best I can tell HA needs the `tasmota/discovery/*` messages - but I can't figure out which Tasmota command (if any) I can use to publish those again, or maybe even make them retained messages?

Any thoughts/ideas would be appreciated.

I have not rebooted most of the devices yet so I have some devices I can test and debug - but keen to get them back online as soon as practical.

Thanks!

**Comment by sillyfrog** [comment, marked as answer]
Looking further, I came across issue https://github.com/arendst/Tasmota/issues/8753

**ABORT ABORT!!!**

The issue is my Mosquitto MQTT server. The update to v2 came through in docker - and the new version "helpfully" does _not_ write retained messages to disk - so after a reboot it was not retaining messages. But when doing quick testing, it was retaining messages.

So that's annoying, but at least explains why I couldn't find the issue elsewhere.

**Reply by sfromis** [reply]
My view is that it is a bad plan to run Mosquitto (or another MQTT broker) under Home Assistant, to be torn down by a crash or planned restart. Even if not using HA, my Mosquitto broker is only restarted once or twice a year. I like smooth sailing :-)

**Reply by sillyfrog** [reply]
@sfromis Agreed (that's what I try to do) :) 

In this instance the whole server went down (brown out I think) - I have been having weird issues for a while that I think are the result of the Mosquitto upgrade. Ever since `stable` was v2 for mosquitto, a number of defaults changed, including persistence (persistence is now `false` by default, which I just figured out) - so any restart/upgrade of Mosquitto caused this to happen (but it was only noticeable after HA restarted and relied on the retained MQTT messages - so I was looking in the wrong place). 

I have been testing further now and my config changes look like they have fixed the issue (ie: Mosquitto is actually retaining messages and writing them to disk). Fingers crossed.

**Reply by sfromis** [reply]
I've been postponing upgrading Mosquitto to v2 till making sure to grok the differences and effects. Anyway, my small UPS ix serving my fiber box, NAS and smarthome server.
